crates/ruma-identity-service-api/src/invitation/sign_invitation_ed25519/v2.rs

```diff
@@ -0,0 +1,60 @@
+//![POST /_matrix/identity/v2/sign-ed25519](https://matrix.org/docs/spec/identity_service/r0.3.0#post-matrix-identity-v2-sign-ed25519)
+
+use ruma_api::ruma_api;
+use ruma_identifiers::{Signatures, UserId};
+
+ruma_api! {
+    metadata: {
+        description: "Sign invitation details.",
+        method: POST,
+        name: "sign_invitation_ed25519",
+        path: "/_matrix/identity/v2/sign-ed25519",
+        authentication: AccessToken,
+        rate_limited: false,
+    }
+
+    request: {
+        /// The Matrix user ID of the user accepting the invitation.
+        pub mxid: UserId,
+
+        /// The token from the call to store-invite.
+        pub token: &'a str,
+
+        /// The private key, encoded as unpadded base64.
+        pub private_key: &'a str,
+    }
+
+    response: {
+        /// The Matrix user ID of the user accepting the invitation.
+        pub mxid: UserId,
+
+        /// The Matrix user ID of the user who sent the invitation.
+        pub sender: UserId,
+
+        /// The signature of the mxid, sender and token.
+        pub signatures: Signatures<String, String>,
```

Comment:
Do we not have better types to use here?

Reply:
Would a `Signatures<String, KeyNameBox>` be better?
I'm not sure if I'm understanding the spec right, but it seems that [ServerSignatures](https://docs.ruma.io/ruma_identifiers/type.ServerSignatures.html) would possibly fit better? I am unsure though since the example in the spec isn't as clear as for my previous PR: I don't know if 'my.id.server' stands for 'my.UserID.server' or 'my.identity.server'. (Thinking about it now, I think it's more likely to be the latter)

Reply:
Yes, that would be better. Not sure about the first `String`, let's hear @iinuwa's opinion.

Reply:
Yeah, it's not clear, but the example looks like a server name, so it's reasonable to use `ServerSignatures` here as well.

@Frinksy, can you also create an issue on the [Matrix spec repo](https://github.com/matrix-org/matrix-doc/issues) asking for clarification on this field?

Reply:
I created an issue [here](https://github.com/matrix-org/matrix-doc/issues/3242).

Reply:
Thanks!